src/pages/preferences/preferencesTabs/PublicKeysTab.tsx

```diff
@@ -1,3 +1,128 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { Table, Skeleton, Popconfirm } from "antd";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import Button from "@leafygreen-ui/button";
+import Icon from "@leafygreen-ui/icon";
+import styled from "@emotion/styled";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  GetMyPublicKeysQuery,
+  GetMyPublicKeysQueryVariables,
+  RemovePublicKeyMutation,
+  RemovePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { REMOVE_PUBLIC_KEY } from "gql/mutations";
 
-export const PublicKeysTab: React.FC = () => <div>Manage Public Keys</div>;
+export const PublicKeysTab: React.FC = () => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [tableData, setTableData] = useState([]);
+  const { data: myKeysData, loading: loadingMyPublicKeys } = useQuery<
+    GetMyPublicKeysQuery,
+    GetMyPublicKeysQueryVariables
+  >(GET_MY_PUBLIC_KEYS, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error fetching your public keys: ${error.message}`
+      );
+    },
+  });
+  const [removePublicKey, { loading: loadingRemovePublicKey }] = useMutation<
+    RemovePublicKeyMutation,
+    RemovePublicKeyMutationVariables
+  >(REMOVE_PUBLIC_KEY, {
+    onCompleted(removeData) {
+      setTableData(removeData.removePublicKey);
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error removing the public key: ${error.message}`
+      );
+    },
+  });
+
+  useEffect(() => {
+    setTableData(myKeysData?.myPublicKeys ?? []);
+  }, [myKeysData]);
+
+  const columns = [
```

Comment:
i don't see any filters in the designs for this table, therefore the columns data can be outside of the component

Reply:
ah, sorting does require the context of the component function body. 👍